### PR TITLE
plan/validator: add grammar check of constraints when create table

### LINKED ISSUE #2127: Primary key can't be duplicated
Please answer these questions before submitting your issue. Thanks!

1. What version of Go are you using (`go version`)?
go version go1.7.1 darwin/amd64

2. What operating system and processor architecture are you using (`go env`)?
GOARCH="amd64"
GOBIN="/Users/xuhuaiyu/Development/GOPATH/bin"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/xuhuaiyu/Development/GOPATH"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/k6/ywdpkc6d49s5s0xm0r50hl9h0000gn/T/go-build952037203=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"

3. What did you do?
create table t (a int key, b int, primary key(a));

4. What did you expect to see?
ERROR 1068 (42000): Multiple primary key defined

5. What did you see instead?
Query OK, 0 rows affected (0.00 sec)

### PR DESCRIPTION
**Pull request by XuHuaiyu**
fix #2127 and #2128 

Check whether exists duplicate cols in constraints of CreateTableStmt, 
and whether exists duplicate primary keys.

PTAL @coocood @shenli @hanfei1991 @zimulala

### CODE REVIEW

File: infoschema/infoschema.go [hunk 1]
```diff
@@ -298,6 +300,8 @@ const (
 	codeBadTable       = 1051
 	codeColumnExists   = 1060
 	codeIndexExists    = 1831
+
```
Comment (zimulala) [inline infoschema/infoschema.go]: Remove this empty line.

File: plan/validator_test.go [hunk 2]
```diff
@@ -54,9 +54,14 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 		{"create table t(id int auto_increment) ENGINE=MYISAM", true, nil},
 		{"create table t(a int primary key, b int, c varchar(10), d char(256));", true, errors.New("Column length too big for column 'd' (max = 255); use BLOB or TEXT instead")},
 		{"create index ib on t(b,a,b);", true, errors.New("[schema:1060]Duplicate column name 'b'")},
-		{"create table t (a int, b int, key(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
-		{"create table t(c1 int not null primary key, c2 int not null primary key)", true, errors.New("Multiple primary key defined")},
 		{"alter table t add index idx(a, b, A)", true, errors.New("[schema:1060]Duplicate column name 'A'")},
+		{"create table t (a int, b int, index(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
```
Comment (zimulala) [inline plan/validator_test.go]: Add the test about "create table t (a int key, b int, primary key(a));".

Comment (XuHuaiyu) [inline plan/validator_test.go]: line64 is a test case like this. @zimulala

File: plan/validator.go [hunk 3]
```diff
@@ -197,21 +197,30 @@ func (v *validator) checkCreateTableGrammar(stmt *ast.CreateTableStmt) {
...
+			err := checkDuplicateColumnName(constraint.Keys)
+			if err != nil {
+				v.err = err
+				return
+			}
+		case ast.ConstraintPrimaryKey:
+			if countPrimayKey > 0 {
+				v.err = infoschema.ErrMultiplePriKey
+				return
+			}
```
Comment (coocood) [inline plan/validator.go]: Should update `countPrimaryKey`.

File: plan/validator_test.go [hunk 4]
```diff
@@ -62,6 +62,7 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 		{"create table t (a int, b int, unique index(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
 		{"create table t(c1 int not null primary key, c2 int not null primary key)", true, errors.New("[schema:1068]Multiple primary key defined")},
 		{"create table t(c1 int not null primary key, c2 int not null, primary key(c1))", true, errors.New("[schema:1068]Multiple primary key defined")},
+		{"create table t(c1 int not null primary key, c2 int not null, primary key(c1), primary key(c2))", true, errors.New("[schema:1068]Multiple primary key defined")},
```
Comment (coocood) [inline plan/validator_test.go]: Have you checked that this test case fails before `countPrimaryKey++` is added.

**Comment by coocood** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
@zimulala PTAL

File: plan/validator_test.go [hunk 5]
```diff
@@ -54,9 +54,15 @@ func (s *testValidatorSuite) TestValidator(c *C) {
...
-		{"create table t(c1 int not null primary key, c2 int not null primary key)", true, errors.New("Multiple primary key defined")},
 		{"alter table t add index idx(a, b, A)", true, errors.New("[schema:1060]Duplicate column name 'A'")},
+		{"create table t (a int, b int, index(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
+		{"create table t (a int, b int, key(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
+		{"create table t (a int, b int, unique(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
+		{"create table t (a int, b int, unique key(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
+		{"create table t (a int, b int, unique index(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
+		{"create table t(c1 int not null primary key, c2 int not null primary key)", true, errors.New("[schema:1068]Multiple primary key defined")},
+		{"create table t(c1 int not null primary key, c2 int not null, primary key(c1))", true, errors.New("[schema:1068]Multiple primary key defined")},
```
Comment (hanfei1991) [inline plan/validator_test.go]: mysql> create table t(c1 int not null primary key, c2 int not null, primary key(c1);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1

Comment (XuHuaiyu) [inline plan/validator_test.go]: The sql lacks a close parenthesis and the end.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**